servicetalk-encoding-api/src/main/java/io/servicetalk/encoding/api/AbstractContentCodec.java

```diff
@@ -30,20 +31,20 @@ public final CharSequence name() {
     }
 
     @Override
-    public boolean equals(final Object o) {
+    public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (o == null || getClass() != o.getClass()) {
```

Comment:
comparing the exact class and instanceof is not quite the same. The existing implementation would declare different sub-classes to be non-equal where this implementation might declare them equivalent. There is also the danger of non-reflexive equals() if one of the sub-classes declares an equals() method that matches specifically against their own class.

I would encourage you to leave it getClass() != o.getClass() unless you can make the equals() method final.

Reply:
good point!
this class is moved in #1376. let's handle it there and as part of #1386

Reply:
here is the argument against `getClass()` : https://errorprone.info/bugpattern/EqualsGetClass